src/build/resources.js

```diff
@@ -0,0 +1,55 @@
+const path = require("path");
+const fs = require("fs").promises;
+const Viz = require("viz.js");
+const vizRenderOpts = require("viz.js/full.render.js");
+
+const COPY_FILES_PATTERN = /\.(jpg|jpeg|png|gif|ms)/;
+const VIZ_RENDER_PATTERN = /\.(dot|neato|fdp|sfdp|twopi|circo)/;
+
+async function buildResources(pageIndex, buildOpts) {
+  await Promise.all(Object.values(pageIndex.pages).map(async (page) => {
+    const files = await fs.readdir(page.dirPath, {encoding: "utf8"});
+
+    //ensure all localized paths exist before we start copying/generating files
+    await Promise.all(page.langs.map(async (lang) => {
+      const localizedPageOutputDir = path.join(buildOpts.outputDir, page.localizedPaths[lang]);
+      await fs.mkdir(localizedPageOutputDir, {recursive: true});
+    }));
+
+    await Promise.all(files.map(async (filePath) => {
+      const srcPath = path.join(page.dirPath, filePath);
+      const {ext, base, name} = path.parse(srcPath);
+      let destLangs = page.langs;
+
+      const singleLangMatch = name.match(/^.*_(\w{2})$/);
```

Comment:
Recall that a page can be built to multiple directories because URLs can be localized. If a resource name ends with a language, e.g. `example_es.jpg` or `example_en.jpg`, then it will only end up in that build of the page. Otherwise the resource is copied to each output directory.